operator/cluster/upgrade.go

```diff
@@ -28,512 +28,692 @@ import (
-	}
+	// set pgouser to match the default configuration currently in use after the Operator upgrade
+	pgcluster.ObjectMeta.Labels[config.LABEL_PGOUSER] = parameters[config.LABEL_PGOUSER]
 
-	// determine if pgbadger is enabled for the cluster using label "crunchy-pgbadger"
-	badgerEnabled, err := strconv.ParseBool(cluster.Labels[config.LABEL_BADGER])
-	if err != nil {
-		return "", err
-	} else if badgerEnabled {
-		badgerContainer := patchDeploymentContainers{
-			Name:  "pgbadger",
-			Image: ccpImagePrefix + "/" + pgBadgerCCPImage + ":" + ccpImageTag,
-		}
-		// see if the image name is overridden
-		if operator.ContainerImageOverrides[config.CONTAINER_IMAGE_CRUNCHY_PGBADGER] != "" {
-			badgerContainer.Image = operator.ContainerImageOverrides[config.CONTAINER_IMAGE_CRUNCHY_PGBADGER]
-		}
-		containersToPatch = append(containersToPatch, badgerContainer)
+	// if the exporter port is not set, set to the configuration value for the default configuration
+	if pgcluster.Spec.ExporterPort == "" {
+		pgcluster.Spec.ExporterPort = operator.Pgo.Cluster.ExporterPort
 	}
 
-	// check pgo.yaml to see if the crunchyadm sidecar is enabled for the current pgo installation
-	if operator.Pgo.Cluster.EnableCrunchyadm {
-		crunchyadmContainer := patchDeploymentContainers{
-			Name:  "crunchyadm",
-			Image: ccpImagePrefix + "/" + crunchyadmCCPImage + ":" + ccpImageTag,
-		}
-		// see if the image name is overridden
-		if operator.ContainerImageOverrides[config.CONTAINER_IMAGE_CRUNCHY_ADMIN] != "" {
-			crunchyadmContainer.Image = operator.ContainerImageOverrides[config.CONTAINER_IMAGE_CRUNCHY_ADMIN]
-		}
-		containersToPatch = append(containersToPatch, crunchyadmContainer)
+	// if the pgbadger port is not set, set to the configuration value for the default configuration
+	if pgcluster.Spec.PGBadgerPort == "" {
+		pgcluster.Spec.PGBadgerPort = operator.Pgo.Cluster.PGBadgerPort
 	}
 
-	// create the json structure required to patch (i.e. update) the images defined in
-	// containersToPatch
-	patchDeployPodSpec := patchDeploymentPodSpec{
-		Containers: containersToPatch,
-	}
-	patchDeployTemplate := patchDeploymentTemplate{
-		PatchDeploymentPodSpec: patchDeployPodSpec,
-	}
-	patchDeploySpec := patchDeploymentSpec{
-		PatchDeploymentTemplate: patchDeployTemplate,
-	}
-	patchImgName := patchImageName{
-		PatchDeploymentSpec: patchDeploySpec,
+	// ensure that the pgo-backrest label is set to 'true' since pgbackrest is required for normal
+	// cluster operations in this version of the Postgres Operator
+	pgcluster.ObjectMeta.Labels[config.LABEL_BACKREST] = "true"
+
+	// add a label with the PGO version upgraded from and to
+	pgcluster.Annotations[config.ANNOTATION_UPGRADE_INFO] = "From_" + oldpgoversion + "_to_" + parameters[config.LABEL_PGO_VERSION]
+	// update the "is upgraded" label to indicate cluster has been upgraded
+	pgcluster.Annotations[config.ANNOTATION_IS_UPGRADED] = "true"
+
+	// set the default memory resource values
+	setMemoryResources(pgcluster)
```

Comment:
We'd also need to set CPU resources, unless I'm missing that?

Reply:
I set memory because when I was testing against 'normally' created clusters, I saw that the memory values were set in the CRDs. I didn't see similar CPU resource values set in the CRD, so I didn't add them to try to mimic the default behavior. If the CPU resources should be added, I'm happy to put those in as well.

Reply:
Yes, the CPU value is not set unless specified, but if it is, it is set in the CRD, e.g.:

https://github.com/CrunchyData/postgres-operator/blob/master/apiserver/clusterservice/clusterimpl.go#L1024-L1030